### ISSUE TITLE
Psalm should not complain when foreach() uses keys only

### ISSUE THREAD

**Issue by BenMorel** [vimeo/psalm#4601]
When iterating an `Iterator` with `foreach()`, we may sometimes only use the yielded keys, and not the values. Example:

```php
$phar = new \PharData('path/to/archive');

foreach ($phar as $path => $file) {
    echo "$path\n";
}
```

In this case, Psalm complains twice:

> INFO: MixedAssignment - 5:28 - Unable to determine the type that $file is being assigned to
> INFO: UnusedVariable - 5:28 - $file is never referenced or the value is not used

https://psalm.dev/r/3560f06057

None of these should be reported if `$file` is unused, IMO:

- `MixedAssignment`: we don't care if Psalm cannot infer the type, as the variable not used anywhere
- `UnusedVariable`: `foreach ()` forces us to give a variable name for the value, even if we don't use it; and `array_keys()` is not an alternative with iterators, so I would consider this a legitimate use and not report an issue here.

**Comment by muglug** [comment]
Prefix with `_`: https://psalm.dev/r/e770d79e4f

**Comment by BenMorel** [comment]
Thank you, @muglug!
